hyperspace/solana/src/lib.rs

```diff
@@ -555,7 +557,7 @@ impl IbcProvider for Client {
 			ibc::core::ics24_host::identifier::PortId,
 		)>,
 	) {
-		todo!()
```

Comment:
Why do we need a lock?

Reply:
because we are using a `Arc<Mutex<>>`

Reply:
Why?

Reply:
it was used in cosmos, so i thought it might be needed when we implement the methods further

Reply:
I don't think we need to use a lock in Solana